worlds/witness/player_logic.py

```diff
@@ -794,23 +846,124 @@ def determine_unrequired_entities(self, world: "WitnessWorld") -> None:
             # Jungle Popup Wall Panel
         }
 
+        # In panel hunt, all panels are game, so all panels need to be reachable (unless disabled)
+        if goal == "panel_hunt":
+            for entity_hex in is_item_required_dict:
+                if static_witness_logic.ENTITIES_BY_HEX[entity_hex]["entityType"] == "Panel":
+                    is_item_required_dict[entity_hex] = True
+
         # Now, return the keys of the dict entries where the result is False to get unrequired major items
         self.ENTITIES_WITHOUT_ENSURED_SOLVABILITY |= {
             item_name for item_name, is_required in is_item_required_dict.items() if not is_required
         }
 
-    def make_event_item_pair(self, entity_hex: str) -> Tuple[str, str]:
-        """
-        Makes a pair of an event panel and its event item
-        """
-        action = " Opened" if self.REFERENCE_LOGIC.ENTITIES_BY_HEX[entity_hex]["entityType"] == "Door" else " Solved"
+    def pick_panel_hunt_panels(self, world: "WitnessWorld"):
+        disallowed_entities_for_panel_hunt = {
+            "0x03629",  # Tutorial Gate Open, which is the panel that is locked by panel hunt
+            "0x03505",  # Tutorial Gate Close (same thing)
+            "0x3352F",  # Gate EP (same thing)
```

Comment:
Does this need to be here, it's not a panel?

Reply:
Generic Entity Hunt foresight